### PR TITLE
Add namespace support and add generate skeleton migration file

### PR DESCRIPTION
**Pull request by baselbj**

### CODE REVIEW

File: Commands/MakeModel.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ public function run(array $params = [])
 
         $overwrite = (bool)CLI::getOption('f');
 
-        $destination = $this->determineOutputPath('Models').$name.'.php';
+        $destination = $this->determineOutputPath('Models', $this->opthions['namespace']).$name.'.php';
```
Comment (lonnieezell) [inline Commands/MakeModel.php]: You have a typo here: `options`, not `opthions`.

**Comment by lonnieezell** [comment]
Looks pretty good so far. As long as we remember to document the -n option :)

One thing, though, is that the Migration command needs to get smarter. We should be able to provide an existing database table, or check if the table exists, and build a new migration, or the skeleton code for changing a table if we know enough.

I have done something similar previously, in Sprint that you can [use as inspiration](https://github.com/ci-bonfire/Sprint/tree/develop/myth/_generators/Migration).

**Comment by baselbj** [comment]
I agree about migration, this is the first step and I am planning to do the smarter one. as for the documentation I am now planning to add the (help feature) for each command which will display:
1) usage
2) description 
3) options

I think this can be by adding a new help function to commands ... maybe creating a vulcanBaseCommand that extends the baseCommand and add the help feature to it.

What do you think?

**Comment by lonnieezell** [comment]
Absolutely yes on the help feature. :) that was something I've had in mind that I need to get back to and implement, but haven't yet. And support for that is something that should be handled at the core CodeIgniter level, honestly.

**Comment by baselbj** [comment]
yes I think I will pull request to Codeigniter to fix this

**Comment by baselbj** [comment]
I will update this

**Comment by baselbj** [comment]
Later I will complete make:model missing functionality and start working on create migration from database

File: Commands/MakeCommand.php [hunk 2]
```diff
@@ -22,13 +22,40 @@ class MakeCommand extends BaseCommand
...
+     * @var string
+     */
+    protected $usage = 'make:command [command_name] [Options]';
+
+    /**
+     * the Command's Arguments
+     *
+     * @var array
+     */
+    protected $arguments = array(
```
Comment (lonnieezell) [inline Commands/MakeCommand.php]: While there's no real style guide for this project, I would prefer to use short array syntax and, in general, keep it close to CI4's styles.

File: Commands/MakeCommand.php [hunk 3]
```diff
@@ -50,12 +77,12 @@ public function run(array $params=[])
         $view = 'Command/Command';
 
         $data = [
-          'namespace' => 'Codeigniter\Commands',
+          'namespace' => is_null(CLI::getOption('n')) ? 'App' : CLI::getOption('n'),
```
Comment (lonnieezell) [inline Commands/MakeCommand.php]: PHP 7's new short syntax for this would be a nice addition: 

    'namespace' => CLI::getOption('n') ?? 'App';

Comment (lonnieezell) [inline Commands/MakeCommand.php]: PHP 7's new short syntax for this would be a nice addition: 

    'namespace' => CLI::getOption('n') ?? 'App';

**Comment by lonnieezell** [comment]
There's a couple of style tweaks I'd like to see implemented but everything else looks good. 

In the future, though, please try to keep separate tasks in different PR's. For instance, namespace support would have been one, and the new generator could have been another. Just keeps things a little easier to work with.

**Comment by baselbj** [comment]
I agree with you but some times I feel excited to delver more things. The other thing is that I don't know how to open and deal with multiple pr's on one repo at the same time ... I am so much new in this github world ... I will search for a tutorial about this.

File: Commands/MakeCommand.php (hunk 2 above)
Comment (baselbj) [reply Commands/MakeCommand.php]: I think we should use the same CI4 style

File: Commands/MakeController.php [hunk 4]
```diff
@@ -82,7 +82,7 @@ public function run(array $params=[])
             : 'Controller/SimpleController';
 
         $data = [
-            'namespace' => is_null(CLI::getOption('n')) ? 'App' : CLI::getOption('n'),
+            'namespace' => 'namespace' => CLI::getOption('n') ?? 'App',
```
Comment (lonnieezell) [inline Commands/MakeController.php]: Oops! Looks like you got some duplicates going on here....

**Comment by lonnieezell** [comment]
It's relatively simple to split the changes up - keep one branch for each major feature that you're working on , and that will do it!

File: Commands/MakeController.php (hunk 4 above)
Comment (baselbj) [reply Commands/MakeController.php]: sorry a few days with few sleeping hours :)

**Pull request merged**